### PR TITLE
NestedFolders: Add empty states for Browse and Search

### PR DESCRIPTION
**Pull request by joshhunt**
Adds empty states for root and folder Browse views, and Search view

image

Part of https://github.com/grafana/grafana/issues/65604

### CODE REVIEW

File: public/app/features/browse-dashboards/components/SearchView.tsx [hunk 1]
```diff
@@ -100,3 +111,12 @@ function assertDashboardViewItemKind(kind: string): DashboardViewItemKind {
...
 }
+
+// const getStyles = (theme: GrafanaTheme2) => ({
+//   noResults: css`
+//     padding: ${theme.v1.spacing.md};
+//     background: ${theme.v1.colors.bg2};
+//     font-style: italic;
+//     margin-top: ${theme.v1.spacing.md};
+//   `,
+// });
```
Comment (tskarhed) [inline public/app/features/browse-dashboards/components/SearchView.tsx:122]: Don't forget about this :wink:

File: public/app/features/browse-dashboards/components/BrowseView.tsx [hunk 2]
```diff
@@ -95,6 +99,27 @@ export function BrowseView({ folderUID, width, height, canSelect }: BrowseViewPr
...
+
+  if (status === 'fulfilled' && flatTree.length === 0) {
+    return (
+      <div style={{ width }}>
+        <EmptyListCTA
+          title={folderUID ? "This folder doesn't have any dashboards yet" : 'No dashboards yet. Create your first!'}
+          buttonIcon="plus"
+          buttonTitle="Create Dashboard"
+          buttonLink={folderUID ? `dashboard/new?folderUid=${folderUID}` : 'dashboard/new'}
+          proTip={folderUID && 'Add/move dashboards to your folder at ->'}
```
Comment (tskarhed) [inline public/app/features/browse-dashboards/components/BrowseView.tsx:114]: I'd skip the arrow
```suggestion
          proTip={folderUID && 'Add/move dashboards to your folder at'}
```

File: public/app/features/browse-dashboards/state/hooks.ts [hunk 3]
```diff
@@ -32,6 +32,16 @@ const selectedItemsForActionsSelector = createSelector(
...
 
+export function useBrowseLoadingStatus(folderUID: string | undefined): 'pending' | 'fulfilled' {
+  return useSelector((wholeState) => {
+    const children = folderUID
+      ? wholeState.browseDashboards.childrenByParentUID[folderUID]
+      : wholeState.browseDashboards.rootItems;
+
+    return children ? 'fulfilled' : 'pending';
+  });
+}
```
Comment (tskarhed) [inline public/app/features/browse-dashboards/state/hooks.ts:43]: Can we get a test for this? I can accept it in another PR if this needs to be merged ASAP

**Review by tskarhed — APPROVED** [review]
These changes can be made in another PR, since this needs to be merged rather quickly.

File: public/app/features/browse-dashboards/components/BrowseView.tsx (hunk 2 above)
Comment (joshhunt) [reply public/app/features/browse-dashboards/components/BrowseView.tsx:114]: I agree - there's a bunch about these EmptyCTA components that I don't like, but I'm going to keep it the same as our current ones for now and we can revisit later.

**Pull request merged**